week-1-ts-ecosystem/w01e03-object-oriented-structure/src/models/product/Product.ts

```diff
@@ -0,0 +1,96 @@
+import {v4} from 'uuid';
+import {IProduct, IProductWithPrice, PRODUCT_TYPE} from './Product.types';
+
+class Product {
+  private readonly _id: string;
+  private _name: string;
+  private _amount: number;
```

Comment:
Pamiętaj że notacja `_` jest umowna.
Więc sam modyfikator `private`  w Typescript jest wiążący i Cię pilnuje.

W tym układzie nie widzę podstaw zastosowania `_` dla nazw zmiennych.
Chyba że potrzebujesz tego w kontekście _setterów / getterów_ - to wtedy tak.

Jednak w JavaScript użycie _setterów / getterów_ ograniczyłbym do przypadków `computed properties`
Na przykład tutaj ładnie wykonane:
```typescript
public set amount(value: number) {}
```
I tam pilnujesz poprawności wartości `value`.

Natomiast `name` mogłoby być po prostu publiczne.
Tak samo `id` jako `public readonly` - daje gwarancje niezmienności

_Hint_ w nowoczesnym JS z tego roku możesz już użyć notacji `#` @DariuszGala .

Przykładowo
```typescript
class Product {
  readonly id: string;
  name: string;
  #amount: number;

  // .....
}
```